### PR TITLE
Update docs and its build action

### PR DESCRIPTION
**Pull request by VJalili**
- [x] This PR updates the docs action to rebuild the website only if any files under the `website` directory are modified;
- [x] Scafold the docs using content from the README and the internally used onboarding document;
- [x] Extend the instructions on building and testing the website for local development.

### CODE REVIEW

File: website/docs/gs/terra.md
Comment (epiercehoffman) [inline website/docs/gs/terra.md:7]: Let's stick with what the README has about Terra for now. The cohort mode workspace is not public yet and should not be referred to on the website.

File: website/docs/gs/dockerfiles.md
Comment (epiercehoffman) [inline website/docs/gs/dockerfiles.md:24]: I think that to a user clicking through the website it is somewhat unclear that they are not responsible for docker images - it is noted in this document, but organizing it with "Getting started" feels confusing. I would suggest creating a "For developers" section and putting this there instead of in "Getting started", or even just creating a README for the `/scripts/docker/` directory of the repo

File: website/docs/modules/make_cohort_vcf.md
Comment (epiercehoffman) [inline website/docs/modules/make_cohort_vcf.md:2]: I know we haven't fixed this on the README yet so it's fine to leave as-is for now but just a heads up that we split MakeCohortVcf into 5 sub-workflows so this will need to change to reflect that soon. How complicated will it be to add new modules or change module names with this website organization?

File: website/docs/gs/wdls.md
Comment (epiercehoffman) [inline website/docs/gs/wdls.md:2]: This content also appears under `/website/docs/modules/index.md`

**Review by epiercehoffman — CHANGES_REQUESTED** [review]
Very cool to see progress towards a website! Thanks for all your work on this.

**Setup:**
I was able to view your branch's version of the website through the localhost successfully.

**Organization:** 
* I am concerned that it will be more difficult to maintain the documentation since there are so many files. Maybe that's unavoidable, but I wanted to raise it as a concern in case there's anything that can be done?
* I think there may be too much info under "Getting Started" - it may be easier to navigate the website if we split up the categories a little more

**Home page:**
The landing/home page seems like it may not give enough information about what GATK-SV actually does, and the three highlights on the landing page are somewhat sweeping claims that I am not sure we can substantiate. What if we turned the home page into a short summary of the pipeline, with some relevant citations, and a quick guide to navigating the website?

**Content:**
I noticed some sections that differ from the README in terms of content. I would prefer at this stage to directly port the entire unaltered README over to the website format and hold off on any further changes to the content for now. Reasoning:
* It's difficult to review content changes when every file is brand new, and I do think we want to review any changes carefully
* It will be complicated to maintain two different versions of the documentation while we are getting the website set up, and more so if they have different information
* Some of the content on this branch is not ready to be public-facing. For example: one section under troubleshooting & the entire cohort mode Terra workspace 
* Some information, such as citations and gCNV training, seems to be missing from this branch of the website, and we don't want to lose track of that
* We want to be working with the User Ed team as we expand our documentation

**Comment by VJalili** [comment]
@epiercehoffman, thanks for the review; I made the changes you suggested. 

> Home page:
> The landing/home page seems like it may not give enough information about what GATK-SV actually does, and the three highlights on the landing page are somewhat sweeping claims that I am not sure we can substantiate. What if we turned the home page into a short summary of the pipeline, with some relevant citations, and a quick guide to navigating the website?

@hab45 and @RCollins13 are working on that. 

> Content:
I noticed some sections that differ from the README in terms of content. I would prefer at this stage to directly port the entire unaltered README over to the website format and hold off on any further changes to the content for now. Reasoning:
> - It's difficult to review content changes when every file is brand new, and I do think we want to review any changes carefully
> - It will be complicated to maintain two different versions of the documentation while we are getting the website set up, and more so if they have different information
> - Some of the content on this branch is not ready to be public-facing. For example: one section under troubleshooting & the entire cohort mode Terra workspace
> - Some information, such as citations and gCNV training, seems to be missing from this branch of the website, and we don't want to lose track of that
> - We want to be working with the User Ed team as we expand our documentation

We tried to start this with content from README and onboarding. Following your suggestions in the comments, I removed all the content from the onboarding doc (if I have not missed any), which should include non-public content. 

> It's difficult to review content changes when every file is brand new, and I do think we want to review any changes carefully

True. Though I am not sure what could be a good alternative to splitting the content into multiple files. Organization in multiple files simplifies the navigation and keeps each page focused on one particular topic. 

> - Some information, such as citations and gCNV training, seems to be missing from this branch of the website, and we don't want to lose track of that

Agree, this is a starting point, and we will incrementally extend and improve. 

> - It will be complicated to maintain two different versions of the documentation while we are getting the website set up, and more so if they have different information
> - We want to be working with the User Ed team as we expand our documentation

I am under the impression that this will largely replace the README so we will not be maintaining two docs. Also, these should be technical docs that we should incrementally improve and extend in faster interactions. I would defer to @mwalker174

File: website/docs/intro.md
Comment (VJalili) [reply website/docs/intro.md]: I removed this section.

File: website/docs/gs/quick_start.md
Comment (VJalili) [reply website/docs/gs/quick_start.md]: Sounds good.

File: website/docs/gs/overview.md
Comment (VJalili) [reply website/docs/gs/overview.md:10]: I added a link to the sections.

File: website/docs/run/_category_.json
Comment (VJalili) [reply website/docs/run/_category_.json]: I remove this section to avoid confusion. We can revisit and add step-by-step docs on how someone can run the pipeline on their own data.

File: website/docs/modules/_category_.json
Comment (VJalili) [reply website/docs/modules/_category_.json:5]: I changed the order according to the README.

File: website/docs/gs/dockerfiles.md
Comment (VJalili) [reply website/docs/gs/dockerfiles.md]: I remove this file, we can revisit and add a section that describes the Docker file and their build process since they are one of the inputs of the pipeline.

File: website/docs/modules/make_cohort_vcf.md
Comment (VJalili) [reply website/docs/modules/make_cohort_vcf.md]: We can add/remove files as needed at this step.

File: website/docs/gs/wdls.md
Comment (VJalili) [reply website/docs/gs/wdls.md:2]: I removed.

File: website/docs/modules/annotate_vcf.md
Comment (VJalili) [reply website/docs/modules/annotate_vcf.md]: I favor having descriptive titles over workflow names, but for this pass, we can keep workflow names in the title.

File: website/docs/execmodes/cohort.md
Comment (VJalili) [reply website/docs/execmodes/cohort.md]: I removed the duplicated sections.

File: website/docs/troubleshooting/inputs.md
Comment (VJalili) [reply website/docs/troubleshooting/inputs.md]: I removed this doc.

File: website/docs/gs/cromwell.md
Comment (VJalili) [reply website/docs/gs/cromwell.md]: I think these came mainly from the onboarding doc, but I agree they should link to their source project. So I removed all these steps.

**Comment by epiercehoffman** [comment]
Thanks for your quick response and updates @VJalili! Quick clarification: 

> I am under the impression that this will largely replace the README so we will not be maintaining two docs.

Agreed, and that is why I want to make sure that the starting website matches the content that's currently in the README, so that we can move over to the website without loss of information, and then once the website is established we can improve the documentation further in one place. 

There are still a number of differences between the README and the website in the current version of your branch so I wanted to clarify this point. What do you think about starting with a direct port of the README over to the website without content changes, so that future content updates are easier to review?

**Comment by VJalili** [comment]
> There are still a number of differences between the README and the website in the current version of your branch so I wanted to clarify this point. What do you think about starting with a direct port of the README over to the website without content changes, so that future content updates are easier to review?

I am not so sure about a direct port. Are you suggesting undoing all the websites, creating a single `.md` file that is a copy-paste of README, and continuing from there?

**Comment by epiercehoffman** [comment]
> Are you suggesting undoing all the websites, creating a single .md file that is a copy-paste of README, and continuing from there?

No, I'm suggesting keeping all the text the same but just reorganizing it into the files needed for the website. So it would involve changes to the organization but save any content changes for later so that we can focus on one thing at a time.

**Comment by VJalili** [comment]
After implementing your suggestions, I think they should be mostly the same. There could be some sentences added/modified, but most detailed descriptions, in particular for the modules, should be the same.

**Review by epiercehoffman — COMMENTED** [review]
Following up on your question earlier today I did another pass through to note sections that are different or missing compared to the README. In addition to the comments throughout the files, the repository structure & troubleshooting sections are also not present in this draft of the website.

File: website/docs/modules/gather_sample_evidence.md
Comment (VJalili) [reply website/docs/modules/gather_sample_evidence.md:7]: I am more in favor of removing them since we have refactored the pipeline, but for this pass, I am happy to add them back.

File: website/docs/modules/evidence_qc.md
Comment (VJalili) [reply website/docs/modules/evidence_qc.md:14]: This was removed as a part response to previous comments. I reverted it.

File: website/docs/gs/input_files.md
Comment (VJalili) [reply website/docs/gs/input_files.md:8]: changed.

File: website/docs/gs/quick_start.md
Comment (VJalili) [reply website/docs/gs/quick_start.md:63]: Added.

File: website/docs/gs/quick_start.md
Comment (VJalili) [reply website/docs/gs/quick_start.md]: I removed the link and the sentence you mentioned. Also elaborated on what does "managed" mean.

File: website/docs/gs/runtime_env.md
Comment (VJalili) [reply website/docs/gs/runtime_env.md]: Removed this paragraph.

File: website/docs/gs/terra.md
Comment (VJalili) [reply website/docs/gs/terra.md]: remove this file.

File: website/docs/gs/wdls.md
Comment (VJalili) [reply website/docs/gs/wdls.md]: Reverted the namings for consistency.

File: website/docs/intro.md
Comment (VJalili) [reply website/docs/intro.md]: updated the link, and extended it with a text suggested by @hab45.

Comment (VJalili) [reply website/docs/intro.md]: I added the link/section in the intro.

File: website/docs/modules/gather_batch_evidence.md
Comment (VJalili) [reply website/docs/modules/gather_batch_evidence.md:15]: added.

File: website/docs/modules/gather_batch_evidence.md
Comment (VJalili) [reply website/docs/modules/gather_batch_evidence.md:13]: Sure, we can update and enrich this in follow-up PRs.

File: website/docs/modules/gather_sample_evidence.md
Comment (VJalili) [reply website/docs/modules/gather_sample_evidence.md:13]: I added that section. That section contains a link to a repo that is private. I am keeping it as they are but should revisit and improve it later.

File: website/docs/modules/regenotype_cnvs.md
Comment (VJalili) [reply website/docs/modules/regenotype_cnvs.md]: fixed.

File: website/docs/modules/gather_sample_evidence.md
Comment (VJalili) [reply website/docs/modules/gather_sample_evidence.md:10]: Added.

**Review by mwalker174 — CHANGES_REQUESTED** [review]
Thank you @VJalili for putting all this together. I have some minor comments about organization, but I was able to easily build and peruse the site locally and it looks nice. I do want to make some more edits, but I'll hold off on that for a future PR.

File: website/docs/gs/runtime_env.md
Comment (VJalili) [reply website/docs/gs/runtime_env.md:16]: Sure, please see the updated format.

File: website/docs/intro.md
Comment (VJalili) [reply website/docs/intro.md:30]: Updated.

File: website/docs/gs/input_files.md
Comment (VJalili) [reply website/docs/gs/input_files.md:57]: done.

File: website/docs/gs/cromwell.md
Comment (VJalili) [reply website/docs/gs/cromwell.md:5]: Updated.

File: website/docs/modules/_category_.json
Comment (VJalili) [reply website/docs/modules/_category_.json:3]: Done.

File: website/docs/gs/quick_start.md
Comment (VJalili) [reply website/docs/gs/quick_start.md:53]: Fixed.

File: website/src/components/HomepageFeatures/index.js
Comment (VJalili) [reply website/src/components/HomepageFeatures/index.js:11]: Updated.

File: website/src/components/HomepageFeatures/index.js
Comment (VJalili) [reply website/src/components/HomepageFeatures/index.js:20]: Updated.

File: website/src/components/HomepageFeatures/index.js
Comment (VJalili) [reply website/src/components/HomepageFeatures/index.js:29]: Updated.

**Comment by VJalili** [comment]
@mwalker174 Thank you for the review. I implemented your suggested changes.

**Review by mwalker174 — APPROVED** [review]
Thanks, looks good!

**Comment by VJalili** [comment]
Thank you, @mwalker174 and @epiercehoffman!

File: website/docs/gs/input_files.md
Comment (epiercehoffman) [inline website/docs/gs/input_files.md:34]: I think this format should match with the format for "Sample IDs must"

File: website/docs/gs/quick_start.md
Comment (epiercehoffman) [inline website/docs/gs/quick_start.md:20]: Not sure if this is helpful to say - if a group has access to a cromwell server then they would know that, but if they don't, this may just be confusing.

File: website/docs/intro.md
Comment (epiercehoffman) [inline website/docs/intro.md:12]:
```suggestion
(Manta, Wham, cnMOPS, GATK-gCNV, MELT). In order to reduce false positives, raw SV 
```

File: website/docs/intro.md
Comment (epiercehoffman) [inline website/docs/intro.md:13]:
```suggestion
are adjudicated and re-genotyped from read evidence, considering all potential 
```

File: website/docs/intro.md
Comment (epiercehoffman) [inline website/docs/intro.md:14]:
```suggestion
sequencing evidence including anomalous paired-end (PE) reads, split reads (SR) crossing 
```

File: website/docs/intro.md
Comment (epiercehoffman) [inline website/docs/intro.md:17]:
```suggestion
has been designed to be deployed in the Google Cloud Platform via the cromwell 
```

File: website/docs/modules/index.md
Comment (epiercehoffman) [inline website/docs/modules/index.md:8]:
```suggestion
consisting of multiple modules to be executed in the following order: 
```

**Review by epiercehoffman — APPROVED** [review]
Thanks for your revisions @VJalili. I made a few more small comments but I recognize we want to get this merged so it's ok with me to merge first and come back to my comments later.

File: website/docs/gs/input_files.md
Comment (VJalili) [reply website/docs/gs/input_files.md:34]: I suppose you mean the heading level; if so, I fixed it.

File: website/docs/gs/quick_start.md
Comment (VJalili) [reply website/docs/gs/quick_start.md:20]: Removed.

File: website/docs/intro.md
Comment (VJalili) [reply website/docs/intro.md:13]: updated.

**Comment by VJalili** [comment]
Thanks, @epiercehoffman! I implemented your suggested changes.

**Pull request merged**
